db_demo/models.py

```diff
@@ -5,3 +5,49 @@
 from django.contrib.auth.models import User
 
 # Your models go here!
+
+
+class Permissions(models.Model):
+    title = models.CharField(max_length=30)
+
+
+class Roles(models.Model):
+    title = models.CharField(max_length=30)
+    permissions = models.ManyToManyField(Permissions)
+    user = models.ManyToManyField(User)
+
+
+class Profile(models.Model):
+    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    body = models.TextField()
+    avatar = models.CharField(max_length=45)
+
+
+class Tags(models.Model):
+    body = models.CharField(max_length=50)
+    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
+
+
+class Categories(models.Model):
+    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    title = models.CharField(max_length=50)
+    parent_id = models.ForeignKey('self', on_delete=models.CASCADE)
```

Comment:
The requirements stated "Categories may have one and only one parent category." May means the field can be blank.